### PR TITLE
Build and run the tests in the Windows build

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
Infrastructure "Build and run the tests in the Windows CI"

#### Purpose of change
We want to shift towards building our Windows releases using Visual Studio, because the cross compiling solution relies on an old gcc.

The nest step towards this is to build and run the tests in the Visual Studio Github Action.

#### Describe the solution
Try to do exactly that.

#### Describe alternatives you've considered
Trying to get someone who actually uses Windows to do this :).

#### Testing
Testing needs to happen here on CI, so opening this as a draft.

#### Additional context
Thanks to @akrieger for getting this CI job working in the first place.

### CODE REVIEW

**Comment by actual-nh** [comment]
You'll also need to remove parts of the paths-ignore (for data/ and tests/).

File: .github/workflows/msvc-full-features.yml [hunk 1]
```diff
@@ -96,6 +96,10 @@ jobs:
             }
         }
 
+    - name: Run tests
+      run: |
+          Cataclysm-test-vcpkg-static.exe --min-duration 0.2 --rng-seed time
```
Comment (jbytheway) [inline .github/workflows/msvc-full-features.yml:101]:
```suggestion
          Cataclysm-test-vcpkg-static-Release-x64.exe --min-duration 0.2 --rng-seed time
```

**Comment by jbytheway** [comment]
@BrettDong seeing some translation test failures. First couple are
```
D:\a\Cataclysm-DDA\Cataclysm-DDA\tests\translation_system_test.cpp(17): FAILED: 

330 REQUIRE( file_exist( path ) ) 

331with expansion: 

332 false 

333 

334------------------------------------------------------------------------------- 

335TranslationManager translates message 

336------------------------------------------------------------------------------- 

337D:\a\Cataclysm-DDA\Cataclysm-DDA\tests\translation_system_test.cpp(55) 

338............................................................................... 

339 

340D:\a\Cataclysm-DDA\Cataclysm-DDA\tests\translation_system_test.cpp(61): FAILED: 

341 CHECK( translated == "батарейка" ) 

342with expansion: 

343 "battery" == "батарейка" 

344 

345-------------------------------------------------------------------------------
```

I'm thinking maybe the `.mo` files are missing? Any thoughts?

**Comment by mqrause** [comment]
I don't know much about all this build stuff, but at least the normal build process with VS requires to execute the bash script `lang/compile_mo.sh` as described in the compilation docs to have all languages available.

**Comment by jbytheway** [comment]
> I don't know much about all this build stuff, but at least the normal build process with VS requires to execute the bash script `lang/compile_mo.sh` as described in the compilation docs to have all languages available.

Thanks for the suggestion.  Let's see if I can make that work...

**Comment by BrettDong** [comment]
Other than compiling real translations with `./lang/compile_mo.sh`, compiling test translation is also needed. The command is in `build.sh` in commit https://github.com/CleverRaven/Cataclysm-DDA/commit/5e4a538073e3edf13beb654f172e5e4bf0d2a506

File: .github/workflows/msvc-full-features.yml [hunk 2]
```diff
@@ -96,6 +94,16 @@ jobs:
             }
         }
 
+    - name: Compile .mo files for localization
+      run: |
+          "C:\Program Files\Git\bin\bash" ./lang/compile_mo.sh all
```
Comment (jbytheway) [inline .github/workflows/msvc-full-features.yml:99]:
```suggestion
          "C:\Program Files\Git\bin\bash" "./lang/compile_mo.sh"
```

File: .github/workflows/msvc-full-features.yml [hunk 3]
```diff
@@ -96,6 +94,16 @@ jobs:
             }
         }
 
+    - name: Compile .mo files for localization
+      run: |
+          "C:\Program Files\Git\bin\bash" "./lang/compile_mo.sh"
```
Comment (jbytheway) [inline .github/workflows/msvc-full-features.yml:99]:
```suggestion
          & "C:\Program Files\Git\bin\bash" "./lang/compile_mo.sh"
```

File: .github/workflows/msvc-full-features.yml (hunk 2 above)
Comment (jbytheway) [inline .github/workflows/msvc-full-features.yml:99]:
```suggestion
          & "C:\Program Files\Git\bin\bash" ./lang/compile_mo.sh all
```

**Comment by jbytheway** [comment]
New windows build is working, and the tests pass! Marked ready for review.

**Pull request merged**